src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -573,9 +573,19 @@ public TResult[] ToArray()
                 }
                 return results;
             }
+
+            public List<TResult> ToList()
+            {
+                var results = new List<TResult>(_source.Length);
```

Comment:
I've an idea for passing `ICollection<T>` instances to the `List<T>` ctor, among other places, which would give the benefit of this and some others, but the best way to do it depends on what we do about #3313.